mlflow/server/js/src/components/ExperimentView.js

```diff
@@ -490,25 +494,34 @@ export class ExperimentView extends Component {
     );
   }
 
-  onSortBy(isMetric, isParam, key) {
-    const sort = this.state.persistedState.sort;
-    this.setSortBy(isMetric, isParam, key, !sort.ascending);
+  onSortBy(orderByKey, orderByAsc) {
+    this.initiateSearch({orderByKey, orderByAsc});
   }
 
-  setSortBy(isMetric, isParam, key, ascending) {
-    const newSortState = {
-      ascending: ascending,
-      key: key,
-      isMetric: isMetric,
-      isParam: isParam
-    };
-    this.setState(
-      {
-        persistedState: new ExperimentViewPersistedState({
-          ...this.state.persistedState,
-          sort: newSortState,
-        }).toJSON(),
-      });
+  initiateSearch({
+      paramKeyFilterInput,
+      metricKeyFilterInput,
+      searchInput,
+      lifecycleFilterInput,
+      orderByKey,
+      orderByAsc,
+    }) {
+    paramKeyFilterInput = (paramKeyFilterInput !== null ?
```

Comment:
I think you want to check `!== undefined` here and elsewhere in this method. Otherwise we end up setting `paramKeyFilter` etc to `undefined` in [onSortBy](https://github.com/mlflow/mlflow/pull/1455/files/65f2e0f7ef74dbb8d24bccc8be1ac897bcb9850c..3e90c9b5e6aa4e77c453e8d94ad6c21f5fecf0e2#diff-fa151380130110b5b1f7ba277afb356aR498) (an absent field in an object evaluates to `undefined` rather than `null`) and therefore overwrite existing metric/param/search filter input here when sorting (you can see this in the current PR by entering some search input & then sorting, the search input goes away).